src/lib/storage/lz4_segment.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <boost/hana/contains.hpp>
+#include <boost/hana/tuple.hpp>
+#include <boost/hana/type.hpp>
+
+#include <type_traits>
+
+#include <array>
+#include <memory>
+
+#include "base_encoded_segment.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class BaseCompressedVector;
+
+template <typename T>
+class LZ4Segment : public BaseEncodedSegment {
+ public:
+  explicit LZ4Segment(const std::shared_ptr<const pmr_vector<char>>& compressed_data,
+                      const std::shared_ptr<const pmr_vector<bool>>& null_values,
+                      const std::shared_ptr<const pmr_vector<size_t>>& offsets, const int compressed_size,
```

Comment:
Can you doc these params, e.g., `offsets`, pls? What is `compressed_size`? Is it equal to `compressed_data.size()`

Reply:
this is still open, isn't it?

Reply:
Yes, compressed_size should equal the size of the vector. We'll remove the compressed_size parameter/member since we can just access the vector size.